utils/config.mli

```diff
@@ -239,6 +244,11 @@ val supports_shared_libraries: bool
 
     @since 4.08 *)
 
+val supports_native_shared_libraries: bool
+(** Whether native shared libraries are supported
```

Comment:
What is the relation between `supports_shared_libraries` and `supports_native_shared_libraries`? Is `supports_shared_libraries` in fact only relevant for bytecode libraries (in which case the comment could say this explicit), or should users check both?

Reply:
Hmm, it's an odd one. As a reference point, the `ocaml` package in opam already exports this variable as `native-dynlink`. `supports_shared_libraries` indicates that the bytecode runtime (and consequently `Dynlink`) can load shared objects. However, in bytecode that also relates to the fact that executables which use C code do not need to be linked with `-custom` (or with `-use-runtime`). In native code, this refers _only_ to native dynlink. So the names are similar, but also subtly different. It makes me wonder if this might be better named `native_dynlink`, but I can see a case for both!